### DISCUSSION TITLE
Element-plus怎样实现下图这种表单自动聚焦的效果？

### DISCUSSION THREAD

**Discussion by lyric-zemin** [element-plus/element-plus#11773] [Q&A]
![20230301143740_rec_](https://user-images.githubusercontent.com/54665054/222063364-03864e6b-4df1-4cef-9657-e201e01ebd36.gif)

**Comment by chenxch** [comment]
双击之后变为编辑模式自动focus么

**Reply by lyric-zemin** [reply]
原本是普通元素，单击后切换为表单元素，可以自动聚焦

**Reply by chenxch** [reply]
目前组件都不支持autofocus，你可以考虑封装个autofocus指令，或者就是要自己获取ref调用focus方法

**Reply by lyric-zemin** [reply]
我测试了focus方法，并不能触发显示下拉列表
[demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtc2VsZWN0IHYtbW9kZWw9XCJ2YWx1ZVwiIGNsYXNzPVwibS0yXCIgcGxhY2Vob2xkZXI9XCJTZWxlY3RcIiBzaXplPVwibGFyZ2VcIiByZWY9XCJzZWxlY3RSZWZcIj5cbiAgICA8ZWwtb3B0aW9uXG4gICAgICB2LWZvcj1cIml0ZW0gaW4gb3B0aW9uc1wiXG4gICAgICA6a2V5PVwiaXRlbS52YWx1ZVwiXG4gICAgICA6bGFiZWw9XCJpdGVtLmxhYmVsXCJcbiAgICAgIDp2YWx1ZT1cIml0ZW0udmFsdWVcIlxuICAgIC8+XG4gIDwvZWwtc2VsZWN0PlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcblxuY29uc3QgdmFsdWUgPSByZWYoJycpXG5jb25zdCBzZWxlY3RSZWYgPSByZWYoKVxuXG5jb25zdCBvcHRpb25zID0gW1xuICB7XG4gICAgdmFsdWU6ICdPcHRpb24xJyxcbiAgICBsYWJlbDogJ09wdGlvbjEnLFxuICB9LFxuICB7XG4gICAgdmFsdWU6ICdPcHRpb24yJyxcbiAgICBsYWJlbDogJ09wdGlvbjInLFxuICB9LFxuICB7XG4gICAgdmFsdWU6ICdPcHRpb24zJyxcbiAgICBsYWJlbDogJ09wdGlvbjMnLFxuICB9LFxuICB7XG4gICAgdmFsdWU6ICdPcHRpb240JyxcbiAgICBsYWJlbDogJ09wdGlvbjQnLFxuICB9LFxuICB7XG4gICAgdmFsdWU6ICdPcHRpb241JyxcbiAgICBsYWJlbDogJ09wdGlvbjUnLFxuICB9LFxuXVxuXG5zZXRUaW1lb3V0KCgpID0+IHtcbiAgc2VsZWN0UmVmLnZhbHVlLmZvY3VzKClcbn0sIDEwMDApXG48L3NjcmlwdD5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==)

**Reply by chenxch** [reply]
[demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtc2VsZWN0IHJlZj1cInNlbGVjdFJlZlwiICB2LW1vZGVsPVwidmFsdWVcIiBjbGFzcz1cIm0tMlwiIHBsYWNlaG9sZGVyPVwiU2VsZWN0XCIgc2l6ZT1cImxhcmdlXCIgPlxuICAgIDxlbC1vcHRpb25cbiAgICAgIHYtZm9yPVwiaXRlbSBpbiBvcHRpb25zXCJcbiAgICAgIDprZXk9XCJpdGVtLnZhbHVlXCJcbiAgICAgIDpsYWJlbD1cIml0ZW0ubGFiZWxcIlxuICAgICAgOnZhbHVlPVwiaXRlbS52YWx1ZVwiXG4gICAgLz5cbiAgPC9lbC1zZWxlY3Q+XG4gIDxlbC1idXR0b24gQGNsaWNrPVwiaGFuZGxlRm9jdXNcIj5mb2N1czwvZWwtYnV0dG9uPlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cbmltcG9ydCB7IHJlZiB9IGZyb20gJ3Z1ZSdcblxuY29uc3QgdmFsdWUgPSByZWYoJycpXG5jb25zdCBzZWxlY3RSZWYgPSByZWYobnVsbClcblxuY29uc3Qgb3B0aW9ucyA9IFtcbiAge1xuICAgIHZhbHVlOiAnT3B0aW9uMScsXG4gICAgbGFiZWw6ICdPcHRpb24xJyxcbiAgfSxcbiAge1xuICAgIHZhbHVlOiAnT3B0aW9uMicsXG4gICAgbGFiZWw6ICdPcHRpb24yJyxcbiAgfSxcbiAge1xuICAgIHZhbHVlOiAnT3B0aW9uMycsXG4gICAgbGFiZWw6ICdPcHRpb24zJyxcbiAgfSxcbiAge1xuICAgIHZhbHVlOiAnT3B0aW9uNCcsXG4gICAgbGFiZWw6ICdPcHRpb240JyxcbiAgfSxcbiAge1xuICAgIHZhbHVlOiAnT3B0aW9uNScsXG4gICAgbGFiZWw6ICdPcHRpb241JyxcbiAgfSxcbl1cbmZ1bmN0aW9uIGhhbmRsZUZvY3VzKCkge1xuICBzZWxlY3RSZWYudmFsdWUudG9nZ2xlTWVudSgpXG59XG48L3NjcmlwdD5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==)  不同组件可能方式不一样
